### ISSUE TITLE
Create webhook without needing to import a repo

### ISSUE THREAD

**Issue by tranceporter** [harness/terraform-provider-harness#1026]
Hi there,

### Terraform Version
1.7.3

### Affected Resource(s)
Please list the resources as a list, for example:
- harness_platform_repo_webhook
- harness_platform_repo
- harness_platform_triggers

### Terraform Configuration Files
```hcl

data "harness_platform_connector_vault" "dev_vault_connector" {
  identifier = "vaultconnectordev"
  org_id     = var.harness_org_id
}

resource "harness_platform_secret_text" "github_webhook" {
  identifier  = "secretgithubwebook"
  name        = "secret-github-webhook"
  description = "GitHub webhook secret used to authenticate webook call with Harness Trigger" #https://developer.harness.io/docs/platform/triggers/triggers-reference/#configure-secret
  org_id      = var.harness_org_id
  tags        = var.harness_tags

  secret_manager_identifier = data.harness_platform_connector_vault.dev_vault_connector.identifier
  value_type                = "Reference"
  value                     = "harness/harness-trigger-secret"
}

resource "harness_platform_repo" "github_repo" {
  identifier     = var.pipeline.trigger.identifier
  name           = var.pipeline.repo_name
 org_id          = var.org_id
  project_id      = var.project_id
  default_branch = "main"
  description    = var.pipeline.repo_name
  source {
    repo = "${var.org}/${var.pipeline.repo_name}"
    type = "github"
  }
}

resource "harness_platform_repo_webhook" "github_webhook" {
  identifier      = format("%sGithubWebhook", var.pipeline.trigger.identifier)
  repo_identifier = var.pipeline.repo_name
  description     = format("Github webhook for triggering %s commit", var.pipeline.repo_name)
  insecure        = false
  enabled         = true
  org_id          = var.org_id
  project_id      = var.project_id
  url             = var.harness_webhook_url
  triggers        = [harness_platform_triggers.pipeline_trigger.id]
   secret          = var.github_webhook_secret
}

resource "harness_platform_triggers" "pipeline_trigger" {
  name       = var.pipeline.trigger.name
  identifier = var.pipeline.trigger.identifier
  org_id     = var.org_id
  project_id = var.project_id
  target_id  = harness_platform_pipeline.pipeline.id

  yaml = templatefile("${path.module}/templates/trigger/${var.pipeline.trigger.template_file_name}", {
    identifier         = var.pipeline.trigger.identifier
    name               = var.pipeline.trigger.name
    org_id             = var.org_id
    project_id         = var.project_id
    target_pipeline_id = harness_platform_pipeline.pipeline.id
    input_set_ref      = harness_platform_input_set.inputset.id
    repo_name          = var.pipeline.repo_name
  })
}

# When creating a new resource, there is a potential race-condition
# as the resource comes up.  This resource will introduce
# a slight delay in further execution to wait for the resources to
# complete.
resource "time_sleep" "trigger_setup" {
  depends_on = [
    harness_platform_triggers.pipeline_trigger
  ]

  create_duration  = "15s"
  destroy_duration = "10s"
}
```

### Panic Output
│ Error: 400 Bad Request
│ 
│   with module.platform_project["platform_engineering"].module.pipeline["homehub_odip_infra"].harness_platform_repo_webhook.github_webhook,
│   on modules/pipeline/webhook.tf line 1, in resource "harness_platform_repo_webhook" "github_webhook":
│    1: resource "harness_platform_repo_webhook" "github_webhook" {

### Expected Behavior
Webhook should have been created

### Actual Behavior
Webhook creation failed

### Steps to Reproduce
So my requirement here is to create a trigger, and then automate the creation of corresponding github webhook to work with the trigger. However creating a repo webhook requires a repo identifier. This means I now have to create a repo resource, and use the source block to import a github repo. This frankly feels like an overkill. We should be able to supply a connector ref for github, and the repo name, and this should be enough to create a webhook? 

Secondly, how do we specify the events that trigger the webook? Github has a section for that in the UI, but I cannot map that to the webhook resource. Is that controller by the trigger yaml when we say "spec/type/push"? 

Thirdly, if we want to impose a secret for authenticating the webhook calls, what exactly is the format to supply this? Supplying a simple string seems wrong. So is there a way we can use a harness secret identifier? In our case, the trigger secret is stored in Hashicorp vault, and I use the secret creation resource to reference the vault secret and create a harness secret. I should be allowed to reference that in the webhook. 

Lastly, the repo resource needs username and password when importing an existing repo instead of using a connector_ref. Can this be fixed?

To summarise, I do not seem to be able to fully automate trigger plus corresponding webhook creation. The only way to create GH webhook is to manually create the trigger in the UI, and that supposedly (not tried yet) creates a webhook in Github? I would really like to be able to automate this end to end without having to needlessly import a GH repo into Harness purely for creating a webhook.

**Comment by tranceporter** [comment]
Never mind. If there an issue with the trigger yaml, it does not create the GH webhook. I fixed the trigger yaml,and it immediately created the webhook in GH
